src/solc_0.8/asset/AssetV2.sol

```diff
@@ -3,9 +3,26 @@
 pragma solidity 0.8.2;
 
 import "./ERC1155ERC721.sol";
+import "../catalyst/interfaces/IAssetAttributesRegistry.sol";
+import "./libraries/AssetHelper.sol";
 
 // solhint-disable-next-line no-empty-blocks
 contract AssetV2 is ERC1155ERC721 {
+    AssetHelper.AssetRegistryData private assetRegistryData;
+
+    /// @notice fulfills the purpose of a constructor in upgradeabale contracts
+    function initialize(
+        address trustedForwarder,
+        address admin,
+        address bouncerAdmin,
+        address predicate,
+        uint8 chainIndex,
+        address assetRegistry
+    ) external {
+        initV2(trustedForwarder, admin, bouncerAdmin, predicate, chainIndex);
```

Comment:
I think that this will fail on upgrade.
```
        _checkInit(0);
        _checkInit(1);
        _checkInit(2);
```

Reply:
and you don't need to initialize everything twice.

Reply:
I will let this code as there is no _checkinit in the already deployed version. https://etherscan.io/address/0x7fbf5c9af42a6d146dcc18762f515692cd5f853b#code